From 4c7a42b261a102267e50240f0f5c55d2737f0cd2 Mon Sep 17 00:00:00 2001
From: Luca Coelho <luciano.coelho@intel.com>
Date: Fri, 27 Mar 2020 10:22:08 +0200
Subject: [PATCH] [NOUPSTREAM] iwlwifi: mvm: use 64-bit version of timespec

The getnstimeofday() function was removed in v5.4, being replaced by a
truly 64-bit version.  Replace our only non-upstream usage of
getnstimeofday() with the new ktime_get_real_s64() function.

type=maint
ticket=none

Change-Id: I4643aeebe79f1f48b27eab1c1edeb1d7f396f4cb
Signed-off-by: Luca Coelho <luciano.coelho@intel.com>
Reviewed-on: https://git-amr-3.devtools.intel.com/gerrit/257132
automatic-review: ec ger unix iil jenkins <EC.GER.UNIX.IIL.JENKINS@INTEL.COM>
Tested-by: ec ger unix iil jenkins <EC.GER.UNIX.IIL.JENKINS@INTEL.COM>
x-iwlwifi-stack-dev: 6651bda5fd07ca61729d3075acf13f655fc40690
(cherry picked from commit a68daa3747f0b0643de3207cb597ffc501c510cd)
Signed-off-by: You-Sheng Yang <vicamo.yang@canonical.com>
---
 drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c b/drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c
index 43430589d..1941c86da 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c
@@ -5171,7 +5171,7 @@ static void iwl_mvm_event_bar_rx_callback(struct iwl_mvm *mvm,
 static u32 iwl_mvm_send_latency_marker_cmd(struct iwl_mvm *mvm, u32 msrmnt,
 					   u16 seq, u16 tid)
 {
-	struct timespec ts;
+	struct timespec64 ts;
 	int ret;
 	struct iwl_mvm_marker_rsp *rsp;
 	struct iwl_mvm_marker *marker;
@@ -5182,7 +5182,7 @@ static u32 iwl_mvm_send_latency_marker_cmd(struct iwl_mvm *mvm, u32 msrmnt,
 	u32 cmd_size = sizeof(struct iwl_mvm_marker) +
 		MARKER_CMD_TX_LAT_PAYLOAD_SIZE * sizeof(u32);
 
-	getnstimeofday(&ts);
+	ktime_get_real_ts64(&ts);
 
 	marker = kzalloc(cmd_size, GFP_KERNEL);
 	if (!marker)
-- 
2.25.1

